src/Features/Core/Portable/FindReferences/DefinitionLocation.SymbolDefinitionLocation.cs

```diff
@@ -20,15 +22,15 @@ internal partial class DefinitionLocation
         private sealed class SymbolDefinitionLocation : DefinitionLocation
         {
             private readonly Workspace _workspace;
-            private readonly ProjectId _referencingProjectId;
             private readonly SymbolKey _symbolKey;
+            private readonly string _symbolAssemblyName;
             private readonly ImmutableArray<TaggedText> _originationParts;
 
-            public SymbolDefinitionLocation(ISymbol definition, Project project)
+            public SymbolDefinitionLocation(Solution solution, ISymbol definition)
             {
-                _workspace = project.Solution.Workspace;
-                _referencingProjectId = project.Id;
+                _workspace = solution.Workspace;
                 _symbolKey = definition.GetSymbolKey();
+                _symbolAssemblyName = definition.ContainingAssembly?.Name;
```

Comment:
This seems way too lossy to me - think about cases where the solution has multiple projects that target different versions of the .NET Framework.  They might have completely different assemblies that have the same "Name". Should we be using `Identity` or something stronger instead?